### PR TITLE
[12.0][MIG] users_ldap_populate

### PR DESCRIPTION
**Pull request by Tardo**
Migration of users_ldap_populate from OCA/server-tools 10.0

Cc @tecnativa

### CODE REVIEW

File: users_ldap_populate/models/populate_wizard.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# Copyright 2012 Therp BV (<http://therp.nl>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/gpl.html).
+
+from odoo import models, fields, api
```
Comment (Trivedi-Vacha-SerpentCS) [inline users_ldap_populate/models/populate_wizard.py:4]: Import statement should follow alphabetic order

File: users_ldap_populate/models/populate_wizard.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        'res.company.ldap',
+        'LDAP Configuration'
+    )
+    users_created = fields.Integer(
+        'Number of users created',
+        readonly=True
+    )
+    users_deactivated = fields.Integer(
+        'Number of users deactivated',
+        readonly=True
```
Comment (Trivedi-Vacha-SerpentCS) [inline users_ldap_populate/models/populate_wizard.py:22]: IMHO good to add help on fields.

File: users_ldap_populate/models/users_ldap.py [hunk 3]
```diff
@@ -0,0 +1,184 @@
+# Copyright 2012 Therp BV (<http://therp.nl>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/gpl.html).
+
+import re
+
+from odoo import models, fields, api, _, SUPERUSER_ID
```
Comment (Trivedi-Vacha-SerpentCS) [inline users_ldap_populate/models/users_ldap.py:6]: Import should follow alphabetic order

**Comment by rafaelbn** [comment]
Please @Yajo revew here thanks

File: users_ldap_populate/models/users_ldap.py [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+            known_user_ids.extend(item['no_deactivate_user_ids'])
+            deactivate_unknown &= item['deactivate_unknown_users']
+        return deactivate_unknown, known_user_ids
+
+    def _get_ldap_entry_dicts(self, conf, user_name='*', timeout=60):
+        """Execute ldap query as defined in conf.
+
+        Don't call self.query because it supresses possible exceptions
+        """
+        ldap_filter = filter_format(conf['ldap_filter'] % user_name, ())
```
Comment (hbrunn) [inline users_ldap_populate/models/users_ldap.py:122]: please cherry-pick https://github.com/OCA/server-tools/pull/1690

File: users_ldap_populate/tests/test_users_ldap_populate.py [hunk 5]
```diff
@@ -63,9 +62,10 @@ def test_users_ldap_populate(self):
             'uid': ['fake'],
             'mail': ['fake@fakery.com'],
         })]):
```
Comment (hbrunn) [inline users_ldap_populate/tests/test_users_ldap_populate.py:64]: if you add a second dict here describing to a user that you add before as inactive (demo user will do), you also test the [activating inactive users codepath](https://codecov.io/gh/OCA/server-auth/compare/42b4727d9d0b9fff3dcc6690887168237818b870...e94d80c8cc3ba08efcbc85d8bcae651e1003cb99/diff#D65-71) which probably raises coveragfe enough to let this test pass

**Review by hbrunn — CHANGES_REQUESTED** [review]
please also add

**Review by yajo — APPROVED** [review]

File: users_ldap_populate/tests/test_users_ldap_populate.py (hunk 5 above)
Comment (Tardo) [reply users_ldap_populate/tests/test_users_ldap_populate.py:64]: I can't use demo user... if the user exists but is inactive the code raises this exception: https://github.com/odoo/odoo/blob/12.0/addons/auth_ldap/models/res_company_ldap.py#L202

**Comment by Tardo** [comment]
@pedrobaeza ready to review

File: users_ldap_populate/tests/test_users_ldap_populate.py (hunk 5 above)
Comment (pedrobaeza) [reply users_ldap_populate/tests/test_users_ldap_populate.py:64]: @hbrunn do you agree with the answer?

**Comment by pedrobaeza** [comment]
@Tardo please squash together adjacent translations commits

File: users_ldap_populate/tests/test_users_ldap_populate.py (hunk 5 above)
Comment (hbrunn) [reply users_ldap_populate/tests/test_users_ldap_populate.py:64]: no. Turn on creating users in the demo config then

Comment (pedrobaeza) [reply users_ldap_populate/tests/test_users_ldap_populate.py:64]: @Tardo do you know what to do then? I confess I'm a bit lost with this stuff.

Comment (Tardo) [reply users_ldap_populate/tests/test_users_ldap_populate.py:64]: @hbrunn Can you verify that '_get_or_create_user' can return a "false" value?
- If the user exists and is actived = The user is used
- If the user exists and is deactivated = Exception
- If the user not exists and creation is not enabled = Exception
- If the user not exists and creation is enabled = Copy/Creates a new user

Comment (hbrunn) [reply users_ldap_populate/tests/test_users_ldap_populate.py:64]: the function won't return a falsy value, but that's not the point here. Just turn on the `create_user` flag, then you actually test what this addon does.

File: users_ldap_populate/models/users_ldap.py (hunk 4 above)
Comment (hbrunn) [reply users_ldap_populate/models/users_ldap.py:122]: this still uses just the string formatting, not `filter_format`

**Review by hbrunn — DISMISSED** [review]

File: users_ldap_populate/tests/test_users_ldap_populate.py (hunk 5 above)
Comment (Tardo) [reply users_ldap_populate/tests/test_users_ldap_populate.py:64]: @hbrunn "create_user" is enabled by default: https://github.com/odoo/odoo/blob/12.0/addons/auth_ldap/models/res_company_ldap.py#L34
I don't understand you, sorry.

**Review by pedrobaeza — APPROVED** [review]
We can't wait more for this module, and the concerns are only in the tests part, so I'm going to merge it right now as is, and later PRs can improve that if it's a problem.

**Pull request merged**